cloudflare/resource_cloudflare_custom_hostnames_test.go

```diff
@@ -0,0 +1,146 @@
+package cloudflare
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccCloudflareCustomHostname_Basic(t *testing.T) {
+	t.Parallel()
+	var customHostname cloudflare.CustomHostname
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	resourceName := "cloudflare_custom_hostname." + rnd
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareCustomHostnameDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareCustomHostnameBasic(zoneID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareCustomHostnameExists(resourceName, &customHostname),
+					resource.TestCheckResourceAttr(
+						resourceName, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(
+						resourceName, "hostname", "thisis@cool.com"),
+					resource.TestCheckResourceAttr(
+						resourceName, "ssl.method", "http"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareCustomHostname_Advanced(t *testing.T) {
+	t.Parallel()
+	var customHostname cloudflare.CustomHostname
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	resourceName := "cloudflare_custom_hostname." + rnd
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareCustomHostnameDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareCustomHostnameAdvanced(zoneID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareCustomHostnameExists(resourceName, &customHostname),
+					resource.TestCheckResourceAttr(
+						resourceName, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(
+						resourceName, "custom_origin_server", "serverone.myinfra.com"),
+					resource.TestCheckResourceAttr(
+						resourceName, "ssl.method", "http"),
+					resource.TestCheckResourceAttr(
+						resourceName, "hostname", "thisis@cool.com"),
+					resource.TestCheckResourceAttr(
+						resourceName, "ssl.custom_hostname_settings.tls13", "on"),
```

Comment:
Have you run these integration tests? A [`TypeList`](https://www.terraform.io/docs/extend/schemas/schema-types.html#typelist) is sequentially indice based which means these nested values should have an index in front of them.